src/__Private/LintRun.hack

```diff
@@ -151,10 +151,11 @@ final class LintRun {
         $files[] = $this->getFileForPath($info->getPathname());
       }
     }
-    $results = await Vec\map_async(
-      $files,
-      async $file ==> await $this->lintFileAsync($config, $file),
-    );
+    $results = vec[];
+    foreach ($files as $file) {
+      /* HHAST_IGNORE_ERROR[DontAwaitInALoop] Intentionally run in serial in case of out-of-memory */
+      $results[] = await $this->lintFileAsync($config, $file);
```

Comment:
Is this equally fast as the parallel version? If it is not, would an Async\Semaphore be a middle ground? I'd like to be able to configure the parallelism that I am comfortable with in the linter config. map_with_limited_concurrency_async would have less overhead than a semaphore, but you'd have to write that logic inline.

Reply:
Updated